app/liquid_tags/wikipedia_tag.rb

```diff
@@ -0,0 +1,99 @@
+class WikipediaTag < LiquidTagBase
+  PARTIAL = "liquids/wikipedia".freeze
+
+  def initialize(tag_name, input, tokens)
+    super
+    @data = get_data(input.strip)
+  end
+
+  def render(_context)
+    ActionController::Base.new.render_to_string(
+      partial: PARTIAL,
+      locals: {
+        title: @data[:title],
+        extract: @data[:extract],
+        url: @data[:url]
+      },
+    )
+  end
+
+  private
+
+  def valid_url?(input)
+    input.match?(%r{^https?://([a-z-]+)\.wikipedia.org/wiki/([\S]+)$})
+  end
+
+  def get_data(input)
+    url = ActionController::Base.helpers.strip_tags(input).strip
+    raise StandardError, "Invalid Wikipedia share URL" unless valid_url?(url)
+
+    uri = URI.parse(url)
+    lang = uri.host.split(".", 2).first
+    title = uri.path.split("/").last
+    anchor = uri.fragment
+
+    if anchor
+      parse_page_with_anchor(url, lang, title, anchor)
+    else
+      parse_page(url, lang, title)
+    end
+  end
+
+  def parse_page_with_anchor(url, lang, title, anchor)
+    response = HTTParty.get("https://#{lang}.wikipedia.org/api/rest_v1/page/mobile-sections/#{title}", format: :json)
+    handle_response_error(response, url)
+
+    text, section_title = get_section_contents(response, anchor, url)
+    title = "#{response['lead']['normalizedtitle']} - #{section_title}"
+
+    {
+      title: title,
+      url: url,
+      extract: text_clean_up(text)
+    }
+  end
+
+  def parse_page(url, lang, title)
+    response = HTTParty.get("https://#{lang}.wikipedia.org/api/rest_v1/page/summary/#{title}", format: :json)
+    handle_response_error(response, url)
+
+    {
+      title: response["title"],
+      url: url,
+      extract: response["extract"]
```

Comment:
I noticed how in the response there are two "extracts". Would it make sense to embed `extract_html` here instead of the `extract` ? It looks a bit more readable.

**extract**

![Screenshot_2020-04-19 New Post - DEV(local) Community 👩‍💻👨‍💻](https://user-images.githubusercontent.com/146201/79682978-01937d80-8227-11ea-9223-37c9a884a2b4.png)

**extract_html**

![Screenshot_2020-04-19 New Post - DEV(local) Community 👩‍💻👨‍💻(1)](https://user-images.githubusercontent.com/146201/79682994-1ec84c00-8227-11ea-8bf7-37af6d446d6c.png)